frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -3836,6 +3836,8 @@ const IR::Node* TypeInference::postorder(IR::SwitchStatement* stat) {
             auto lt = getType(c->label);
             if (lt == nullptr)
                 continue;
+            if (auto se = lt->to<IR::Type_SerEnum>())
+                lt = se->type;
```

Comment:
I am not 100% sure with this change but will the following still work:
```
enum bit<4> e
{
  a = 1
}
control c(in e v)
{
  apply{
    switch(v) {
      e.a:
      default: {}
    }
  }
}
```
and will this still be rejected:
```
enum bit<4> e
{
  a = 1
}
control c(in e v)
{
  apply{
    switch(v) {
      1:
      default: {}
    }
  }
}
```

Reply:
I added theses test cases.